apis/telemetry/v1alpha1/telemetry_methods.go

```diff
@@ -0,0 +1,99 @@
+package v1alpha1
+
+import (
+	"errors"
+
+	"reflect"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+	promapi "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// Merge takes an object a and merges another object, b's values into it, overwriting any which conflict.
+func (a *TelemetrySpec) Merge(b *TelemetrySpec) *TelemetrySpec {
+	// TODO: This method is brittle. It must be updated whenever any field is added to `TelemetrySpec``. It
+	// would be best to replace these methods with a generic client side `Merge()` based on the k8s strategic merge
+	// patch logic.
+	out := TelemetrySpec{}
+	if a == nil && b == nil {
+		return &out
+	} else if b == nil {
+		return a
+	} else if a == nil {
+		return b
+	}
+	if b.Prometheus == nil {
+		out.Prometheus = a.Prometheus
+	} else {
+		out.Prometheus = a.Prometheus.Merge(b.Prometheus)
+	}
+	return &out
+}
+
+// Merge takes an object a and merges another object, b's values into it, overwriting any which conflict.
+func (a *PrometheusTelemetrySpec) Merge(b *PrometheusTelemetrySpec) *PrometheusTelemetrySpec {
+	// TODO: This method is brittle. It must be updated whenever any field is added to `PrometheusTelemetrySpec`. It
+	// would be best to replace these methods with a generic client side `Merge()` based on the k8s strategic merge
+	// patch logic.
+	out := PrometheusTelemetrySpec{}
+	if a == nil && b == nil {
+		return nil
+	} else if b == nil {
+		return a
+	} else if a == nil {
+		return b
+	}
+	if a.CommonLabels != nil {
```

Comment:
You could use `utils.MergeMap` here.

Reply:
That wouldn't hurt, done!

Reply:
Well, I thought it was done. Importing utils into the APIs seems to lead to circular dependancies, so this isn't possible unfortunately.